From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Miguel Fuertes <hkfuertes@users.noreply.github.com>
Date: Fri, 25 Sep 2026 09:16:39 +0200
Subject: [PATCH 1/3] android: find bionic libraries with AC_SEARCH_LIBS

Bionic has no separate libpthread/librt; AC_SEARCH_LIBS finds the symbols in libc.

Upstream: not submitted (Android/Bionic only; not needed with glibc/musl)
Signed-off-by: Miguel Fuertes <hkfuertes@users.noreply.github.com>
---
 configure.ac | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/configure.ac b/configure.ac
index 4087ea6..e724af6 100644
--- a/configure.ac
+++ b/configure.ac
@@ -47,7 +47,7 @@ PKG_PROG_PKG_CONFIG([0.9.0])
 
 # Checks for libraries.
 if test "x${with_os}" = xlinux -o "x${with_os}" = xfreebsd ; then
- AC_CHECK_LIB([rt],[clock_gettime], , AC_MSG_ERROR(librt needed))
+ AC_SEARCH_LIBS([clock_gettime], [rt], , AC_MSG_ERROR(librt needed))
 fi
 
 if test "x${with_os}" = xopenbsd ; then
@@ -69,7 +69,7 @@ AC_ARG_WITH([pkg_config],[AS_HELP_STRING([--with-pkg-config],[use pkg-config to
 ##### The following check for the pthreads library doesn't put the compiler into the correct pthread mode
 ##### so we add the -pthread compilation flag in AMFLAGS in the Makefile.am as well.
 
-AC_CHECK_LIB([pthread],[pthread_create], , AC_MSG_ERROR(pthread library needed))
+AC_SEARCH_LIBS([pthread_create], [pthread], , AC_MSG_ERROR(pthread library needed))
 AC_CHECK_LIB([m],[exp], , AC_MSG_ERROR(maths library needed))
 
 if  test "x${with_pkg_config}" = xyes ; then
